pkg/bootstrap/static.go

```diff
@@ -0,0 +1,392 @@
+			UID:          fmt.Sprintf("kubernetes://%s.%s", name, namespace),
+			Operation:    "Check",
+		}
+		opts.Upstreams = []Upstream{*opts.TelemetryUpstream}
+	case "policy":
+		opts.Upstreams = []Upstream{{
+			ListenPort:   15004,
+			UpstreamPort: 9091,
+			GRPC:         true,
+			Service:      fmt.Sprintf("istio-policy.%s.svc.cluster.local", namespace),
+			UID:          fmt.Sprintf("kubernetes://%s.%s", name, namespace),
+			Operation:    "Report",
+		}}
+	}
+	return opts
+}
+
+// BuildBootstrap creates a static config for the sidecar HTTP proxy.
+func BuildBootstrap(opts Options) (*bootstrap.Bootstrap, error) {
+	out := bootstrap.Bootstrap{
+		Admin: bootstrap.Admin{
+			AccessLogPath: "/dev/stdout",
+			Address: core.Address{
+				Address: &core.Address_SocketAddress{
+					SocketAddress: &core.SocketAddress{
+						Address:       "127.0.0.1",
+						PortSpecifier: &core.SocketAddress_PortValue{PortValue: uint32(opts.AdminPort)},
+					},
+				},
+			},
+		},
+		StaticResources: &bootstrap.Bootstrap_StaticResources{
+			Clusters: BuildClusters(opts.Upstreams),
+		},
+	}
+
+	if opts.ZipkinAddress != "" {
+		zipkinAddress, err := toAddress(opts.ZipkinAddress)
+		if err != nil {
+			return nil, err
+		}
+		out.StaticResources.Clusters = append(out.StaticResources.Clusters,
+			v2.Cluster{
+				Name:           ZipkinCluster,
+				ConnectTimeout: 1 * time.Second,
+				Type:           v2.Cluster_STRICT_DNS,
+				LbPolicy:       v2.Cluster_ROUND_ROBIN,
+				Hosts:          []*core.Address{zipkinAddress},
+			})
+		out.Tracing = &trace.Tracing{
+			Http: &trace.Tracing_Http{
+				Name: util.Zipkin,
+				Config: toStruct(&trace.ZipkinConfig{
+					CollectorCluster:  ZipkinCluster,
+					CollectorEndpoint: "/api/v1/spans",
+				}),
+			},
+		}
+	}
+
+	// telemetry bootstrap should route telemetry directly to the local upstream
+	telemetryCluster := TelemetryCluster
+	if opts.TelemetryUpstream == nil {
+		telemetryAddress, err := toAddress(opts.TelemetryAddress)
+		if err != nil {
+			return nil, err
+		}
+		cluster := v2.Cluster{
+			Name:                 TelemetryCluster,
+			ConnectTimeout:       1 * time.Second,
+			Type:                 v2.Cluster_STRICT_DNS,
+			LbPolicy:             v2.Cluster_ROUND_ROBIN,
+			Hosts:                []*core.Address{telemetryAddress},
+			CircuitBreakers:      circuitBreakers,
+			Http2ProtocolOptions: &core.Http2ProtocolOptions{},
+		}
+		if opts.ControlPlaneAuth {
+			cluster.TlsContext = &auth.UpstreamTlsContext{
+				CommonTlsContext: commonTLSContext([]string{opts.TelemetrySAN}),
+			}
+		}
+		out.StaticResources.Clusters = append(out.StaticResources.Clusters, cluster)
+	} else {
+		telemetryCluster = clusterName(*opts.TelemetryUpstream)
+	}
+
+	out.StaticResources.Listeners = BuildListeners(opts.Upstreams, telemetryCluster, opts.ControlPlaneAuth)
+
+	if err := out.Validate(); err != nil {
+		return nil, err
+	}
+
+	return &out, nil
+}
+
+func commonTLSContext(altNames []string) *auth.CommonTlsContext {
```

Comment:
I'd be nice to share this with the core logic here: https://github.com/istio/istio/blob/master/pilot/pkg/networking/plugin/authn/authentication.go#L203

(e.g. have a common method they both defer to)